packages/driver/src/cy/assertions.coffee

```diff
@@ -56,6 +58,35 @@ parseValueActualAndExpected = (value, actual, expected) ->
 
   obj
 
+prepareObjsForDiff = (err) ->
+  if _.isString(err.actual) || _.isString(err.expected)
+    return err
+  ret = {}
+  ret.actual = mochaUtils.stringify(err.actual)
+  ret.expected = mochaUtils.stringify(err.expected)
+  return ret
+
+objToString = Object.prototype.toString
+
+_isSameType = (a, b) ->
+  return objToString.call(a) is objToString.call(b)
+
+_hasExpectedValue = (err) ->
+  err && err.expected isnt undefined
+
+_assertionHasShowDiff = (err) ->
+  err && err.showDiff isnt false
+
+_isExpectedActualSameType = (err) ->
+  _isSameType(err.actual, err.expected)
+
+shouldShowDiff = (err) ->
+  return (
+    _assertionHasShowDiff(err) &&
```

Comment:
I would add `err &&` here and remove `err && ...` from the above two predicates. You are not checking `err && ...` anyway inside `_isExpectedActualSameType` any way